R/summarize_normals.R

```diff
@@ -3,7 +3,11 @@
 # tar_load(gdd_doy_stack_50)
 
 summarize_normals <- function(stack, years = 1991:2020) {
-  stack[[as.character(years)]] |> 
+  
+  years <- as.character(years)
+  years <- years[ which(years %in% names(stack))]
+  
```

Comment:
Could we print a warning when not all the years 1991:2020 are in the stack?  1991-2020 is a defined period, so any data or plots that don't contain that whole range aren't totally correct in this situation.  This would just be a reminder in case you thought you were running all the years or something.

Reply:
Ah, I see. That makes good sense to me. 

Adding 
```
 
  if(!(all(as.character(1991:2020)  %in% names(stack)))) {
    message("stack does not contain all years from 1991:2020")
  }
```

prints a message as the pipeline is running, but also seems to introduce a whole bunch of warnings about 

```
1: In class(object) <- "environment" :
  Setting class(x) to "environment" sets attribute to NULL; result will no longer be an S4 object
```

Do you have more experience adding warnings in targets?

Reply:
Weird!  Not sure what's going on.  If you push these changes I can play around with it.

Reply:
I didn't get this message, so I suspect it might be unrelated to the edits to `summarize_normals()`.